docs/reference/indices/clone-index.asciidoc

```diff
@@ -0,0 +1,137 @@
+the new index.
+
+[float]
+=== How does cloning work?
+
+Cloning works as follows:
+
+* First, it creates a new target index with the same definition as the source
+  index.
+
+* Then it hard-links segments from the source index into the target index. (If
+  the file system doesn't support hard-linking, then all segments are copied
+  into the new index, which is a much more time consuming process.)
+
+* Finally, it recovers the target index as though it were a closed index which
+  had just been re-opened.
+
+[float]
+=== Preparing an index for cloning
+
+Create a new index:
+
+[source,js]
+--------------------------------------------------
+PUT my_source_index
+{
+  "settings": {
+    "index.number_of_shards" : 5
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+In order to clone an index, the index must be marked as read-only,
+and have <<cluster-health,health>> `green`.
+
+This can be achieved with the following request:
+
+[source,js]
+--------------------------------------------------
+PUT /my_source_index/_settings
+{
+  "settings": {
+    "index.blocks.write": true <1>
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+<1> Prevents write operations to this index while still allowing metadata
+    changes like deleting the index.
+
+[float]
+=== Cloning an index
+
+To clone `my_source_index` into a new index called `my_target_index`, issue
+the following request:
+
+[source,js]
+--------------------------------------------------
+POST my_source_index/_clone/my_target_index
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+The above request returns immediately once the target index has been added to
+the cluster state -- it doesn't wait for the clone operation to start.
+
+[IMPORTANT]
+=====================================
+
+Indices can only be cloned if they satisfy the following requirements:
+
+* the target index must not exist
+
+* The source index must have the same number of primary shards as the target index.
+
+* The node handling the clone process must have sufficient free disk space to
+  accommodate a second copy of the existing index.
+
+=====================================
+
+The `_clone` API is similar to the <<indices-create-index, `create index` API>>
+and accepts `settings` and `aliases` parameters for the target index:
+
+[source,js]
+--------------------------------------------------
+POST my_source_index/_clone/my_target_index
+{
+  "settings": {
+    "index.number_of_shards": 5 <1>
+  },
+  "aliases": {
+    "my_search_indices": {}
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[s/^/PUT my_source_index\n{"settings": {"index.blocks.write": true, "index.number_of_shards": "5"}}\n/]
+
+<1> The number of shards in the target index. This must be equal to the
+    number of shards in the source index.
+
+
+NOTE: Mappings may not be specified in the `_clone` request.
```

Comment:
Perhaps mention here that the original indices' mappings will be copied to the new index

Reply:
I've added some words in 10c7b33